### PR TITLE
Fix `*-dark` rule

### LINKED ISSUE #1174: `*-dark`
### Prerequisites

- [X] I have [searched](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

`.border-dark` applies changes for text. Need to see if there's any other class that does these unintended changes. Then need to see which solution to apply here.

### Reduced test cases

Add for example `.border-dark` to an alert with a link inside : link changes its color.

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome, Firefox, Microsoft Edge

### What version of Boosted are you using?

main-branch

### PR DESCRIPTION
**Pull request by louismaximepiton**
Fixes #1174 

### Search

I've searched for every `-dark` class in Boosted and even for `"dark"` word. It appears that every time it is called, this is for a background-color or at least a component that benefits from the rule. The only class that diverged was `.border-dark` that changes the border-color. So I decided to keep the rule and precised that `.border-dark` shouldn't interfere with it anymore. 

My concerns are for Boosted users since this is a general rule that could break existing sites. If you agree, we should at least warn somewhere about this rule and add it to the migration guide. If you think that this rule should be more specific, please let me know.

### Miscellaneous (out of PR)

~~I also discovered that Bootstrap has a `bg-black` that could maybe be removed (which will simplify our utility) since it's not mentioned anywhere nor used. I couldn't spot any breaking change using this, but I added a selector to that rule to handle this specific case.~~
**Edit:** Looks like this bg-black is wanted so let's keep it : https://github.com/twbs/bootstrap/pull/34100

~~The second one is that we have plenty of classes `.a11y-swatch-*` that aren't used yet in `site/assets/scss/colors.scss`. Was wondering if we could use them to have maybe a better accessibility or visibility about this topic. Otherwise, they could be removed easily IMO.~~
**Edit:** Issue created #1297.

### CODE REVIEW

**Comment by julien-deramond** [comment]
Note: we need to check if it works with the new `.text-bg-dark`

File: scss/_root.scss [hunk 1]
```diff
@@ -82,7 +82,8 @@
 }
 
 // Boosted mod
-[class*="-dark"],
+.bg-black,
+[class*="-dark"]:not(.border-dark),
 .bg-secondary {
```
Comment (julien-deramond) [inline scss/_root.scss:87]:
```suggestion
[class*="bg-secondary"] {
```

Apparently there's no `.text-bg-black`.

**Review by julien-deramond — CHANGES_REQUESTED** [review]
This change looks good to me.

As said in the description, I also think we should mention somewhere in the documentation that those rules exist in order to be able to handle dark variants. Not a big fan of the place, but I'd said the most appropriate place is `/docs/5.2/customize/color/`. I let you propose something so that we can iterate on.

FYI, code used to test different use cases (doesn't include cascade of divs with different bg colors):

```html
<div class="container-fluid p-4">
  <div class="row">
    <div class="col-12">
      1. Text
      <br>
      2. <a href="/">Link</a>
      <br>
      3. <code>test</code>
      <br>
      4. <kbd>Ctrl</kbd>
      <br>
      5. <mark>highlight</mark>
      <br>
      6. <pre>pre content</pre>
    </div>
  </div>
</div>

<div class="container-fluid p-4 border-dark border">
  <div class="row">
    <div class="col-12">
      1. Text
      <br>
      2. <a href="/">Link</a>
      <br>
      3. <code>test</code>
      <br>
      4. <kbd>Ctrl</kbd>
      <br>
      5. <mark>highlight</mark>
      <br>
      6. <pre>pre content</pre>
    </div>
  </div>
</div>

<div class="container-fluid p-4 text-dark">
  <div class="row">
    <div class="col-12">
      1. Text
      <br>
      2. <a href="/">Link</a>
      <br>
      3. <code>test</code>
      <br>
      4. <kbd>Ctrl</kbd>
      <br>
      5. <mark>highlight</mark>
      <br>
      6. <pre>pre content</pre>
    </div>
  </div>
</div>

<div class="container-fluid p-4 bg-black">
  <div class="row">
    <div class="col-12">
      1. Text
      <br>
      2. <a href="/">Link</a>
      <br>
      3. <code>test</code>
      <br>
      4. <kbd>Ctrl</kbd>
      <br>
      5. <mark>highlight</mark>
      <br>
      6. <pre>pre content</pre>
    </div>
  </div>
</div>

<div class="container-fluid p-4 bg-dark">
  <div class="row">
    <div class="col-12">
      1. Text
      <br>
      2. <a href="/">Link</a>
      <br>
      3. <code>test</code>
      <br>
      4. <kbd>Ctrl</kbd>
      <br>
      5. <mark>highlight</mark>
      <br>
      6. <pre>pre content</pre>
    </div>
  </div>
</div>

<div class="container-fluid p-4 bg-secondary">
  <div class="row">
    <div class="col-12">
      1. Text
      <br>
      2. <a href="/">Link</a>
      <br>
      3. <code>test</code>
      <br>
      4. <kbd>Ctrl</kbd>
      <br>
      5. <mark>highlight</mark>
      <br>
      6. <pre>pre content</pre>
    </div>
  </div>
</div>

<div class="container-fluid p-4 text-bg-dark">
  <div class="row">
    <div class="col-12">
      1. Text
      <br>
      2. <a href="/">Link</a>
      <br>
      3. <code>test</code>
      <br>
      4. <kbd>Ctrl</kbd>
      <br>
      5. <mark>highlight</mark>
      <br>
      6. <pre>pre content</pre>
    </div>
  </div>
</div>

<div class="container-fluid p-4 text-bg-secondary">
  <div class="row">
    <div class="col-12">
      1. Text
      <br>
      2. <a href="/">Link</a>
      <br>
      3. <code>test</code>
      <br>
      4. <kbd>Ctrl</kbd>
      <br>
      5. <mark>highlight</mark>
      <br>
      6. <pre>pre content</pre>
    </div>
  </div>
</div>
```

**Comment by louismaximepiton** [comment]
> This change looks good to me.
> 
> As said in the description, I also think we should mention somewhere in the documentation that those rules exist in order to be able to handle dark variants. Not a big fan of the place, but I'd said the most appropriate place is `/docs/5.2/customize/color/`. I let you propose something so that we can iterate on.

Tried to place it with Root variables and component explanation.

**Pull request merged**